python/robot_properties_solo/quadruped12wrapper.py

```diff
@@ -21,9 +21,12 @@ def initPhysicsClient():
         p.setPhysicsEngineParameter(fixedTimeStep=dt, numSubSteps=1)
         return physicsClient
 
-    def __init__(self, physicsClient=None):
+    def __init__(self, physicsClient=None, ifrecord = False):
         if physicsClient is None:
             self.physicsClient = self.initPhysicsClient()
+        
+        if ifrecord:
+            p.startStateLogging(p.STATE_LOGGING_VIDEO_MP4, "solo.mp4")
```

Comment:
Mhmm, I would rather go for having a method like

```
robot.start_recording("optional_movie_file_name.mp4")
robot.stop_recording()
```

This allows to reuse the same robot instance to record multiple videos. 

Also, should we add a base "BulletRobot" class in `robot_properties_base`, which has helper methods like this? This way, other robots like solo8 and bolt will get the same functionality. Though this can be done later.

What do you think?

For now, I am fine to land the change

Reply:
I think your suggestion for method names makes more sense. I will try to update the functions accordingly.